api/clients/v2/coretypes/payload.go

```diff
@@ -1,78 +1,47 @@
 package coretypes
 
 import (
-	"fmt"
+	"encoding/binary"
 
 	"github.com/Layr-Labs/eigenda/api/clients/codecs"
 	"github.com/Layr-Labs/eigenda/encoding"
-	"github.com/Layr-Labs/eigenda/encoding/fft"
-	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	"github.com/Layr-Labs/eigenda/encoding/utils/codec"
 )
 
 // Payload represents arbitrary user data, without any processing.
-type Payload struct {
-	bytes []byte
-}
+type Payload []byte
 
-// NewPayload wraps an arbitrary array of bytes into a Payload type.
-func NewPayload(payloadBytes []byte) *Payload {
-	return &Payload{
-		bytes: payloadBytes,
-	}
-}
+// ToEncodedPayload performs the [codecs.PayloadEncodingVersion0] encoding to create an encoded payload.
+func (p Payload) ToEncodedPayload() *EncodedPayload {
+	// Encode payload modulo bn254, and align to 32 bytes
+	encodedData := codec.PadPayload(p)
 
-// ToBlob converts the Payload bytes into a Blob
-//
-// The payloadForm indicates how payloads are interpreted. The form of a payload dictates what conversion, if any, must
-// be performed when creating a blob from the payload.
-func (p *Payload) ToBlob(payloadForm codecs.PolynomialForm) (*Blob, error) {
-	encodedPayload := newEncodedPayload(p)
+	// Calculate the length of the EncodedPayload in symbols (including the header) which has to be a power of 2.
+	encodedDataLenSymbols := uint32(len(encodedData)+encoding.BYTES_PER_SYMBOL-1) / encoding.BYTES_PER_SYMBOL
+	encodedHeaderAndDataLenSymbols := codec.PayloadHeaderSizeSymbols + encodedDataLenSymbols
+	encodedPayloadLenSymbols := encoding.NextPowerOf2(encodedHeaderAndDataLenSymbols)
```

Comment:
Given the redefinition of an encoded payload (always power of 2 length):

1. encoded payload doc needs to be updated to reflect the change
2. would it make sense to create an encoded payload constructor, which asserts that the input bytes are a proper power of 2?